client_update.go

```diff
@@ -86,8 +86,10 @@ func (u *UpdateClient) FetchUpdate(api ApiRequester, url string) (io.ReadCloser,
 	}
 
 	if r.ContentLength < 0 {
+		defer r.Body.Close()
```

Comment:
Because we still want to receive the update ;) otherwise io stream will get closed when fetch update returns.